app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,65 @@
+class OrdersController < ApplicationController
+  before_action :load_categories, only: :index
+  before_action :cart_items, only: :create
+
+  def index
+    return redirect_to root_path if current_user.nil?
+    
+    @pagy, @orders = pagy(load_orders, items: Settings.order.paginates)
+  end
+
+  def create
+    @order = current_user.orders.build(order_params)
+    begin
+      build_order_items_from_cart
+      ActiveRecord::Base.transaction do
+        @order.save!
+        cookies.delete(:cart_items)
+        flash[:success] = "Order successfully placed! Please check your email to track the notification"
+        redirect_to orders_path
+      end
+    rescue ActiveRecord::RecordNotFound => e
+      reload_cart
+      flash[:error] = "Product not found."
+      redirect_back(fallback_location: cart_path)
+    rescue ActiveRecord::RecordInvalid => e
```

Comment:
RecordInvalid ở đây là cái j valid e ? có show msg lỗi cụ thể cái j fail ko?

Reply:
Dạ vậy em in ra e.message có oke không a

Reply:
Dạ vậy em in ra lỗi là e.message có oke không ạ